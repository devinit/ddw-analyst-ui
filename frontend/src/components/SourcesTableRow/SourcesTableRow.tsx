import classNames from 'classnames';
import * as React from 'react';

export interface SourcesTableRowProps {
  classNames?: string;
  onClick: () => void;
  indicator: string;
  indicatorAcronym: string;
  updatedOn: string;
}

export const SourcesTableRow: React.SFC<SourcesTableRowProps> = (props) => {
  return (
    <tr className={props.classNames} onClick={props.onClick} data-testid="sources-table-row">
      <td>
        {props.indicator}
        <span className={classNames('text-uppercase', { 'd-none': !props.indicatorAcronym })}>
          {` (${props.indicatorAcronym})`}
        </span>
      </td>
      <td>{new Date(props.updatedOn).toDateString()}</td>
    </tr>
  );
};
