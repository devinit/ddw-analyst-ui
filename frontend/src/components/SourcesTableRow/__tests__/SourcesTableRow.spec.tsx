/**
 * @jest-environment jsdom
 */
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { fireEvent, render } from 'react-testing-library';
import { SourcesTableRow, SourcesTableRowProps } from '../SourcesTableRow';

const props: SourcesTableRowProps = {
  onClick: jest.fn(),
  count: 1,
  indicator: 'World Bank Indicators',
  indicatorAcronym: 'wbi',
  updatedOn: new Date('August 19, 2018 23:15:30').toISOString()
};

test('renders correctly with the default props', () => {
  const renderer = TestRenderer.create(<SourcesTableRow { ...props }/>).toJSON();

  expect(renderer).toMatchSnapshot();
});

test('responds to click events', () => {
  const table = document.createElement('table');
  const tableBody = document.createElement('tbody');
  table.appendChild(tableBody);
  const { getByTestId } = render(<SourcesTableRow { ...props }/>, {
    container: document.body.appendChild(tableBody)
  });

  fireEvent.click(getByTestId('sources-table-row'));

  expect(props.onClick).toHaveBeenCalled();
});
