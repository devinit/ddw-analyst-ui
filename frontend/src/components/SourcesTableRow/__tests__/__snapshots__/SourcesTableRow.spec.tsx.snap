// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with the default props 1`] = `
<tr
  data-testid="sources-table-row"
  onClick={[MockFunction]}
>
  <td>
    World Bank Indicators
    <span
      className="text-uppercase"
    >
       (wbi)
    </span>
  </td>
  <td>
    Sun Aug 19 2018
  </td>
</tr>
`;
